ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml

```diff
@@ -342,6 +356,240 @@ objects:
           oc wait namespace/openshift-ssh-bastion --for delete --timeout=10m
         }
 
+        function recover-from-etcd-quorum-loss() {
+          mkdir -p ~/.ssh || true
+          cp "${KUBE_SSH_KEY_PATH}" ~/.ssh/id_rsa
+          chmod 0600 ~/.ssh/id_rsa
+          echo "${USER:-default}:x:$(id -u):$(id -g):Default User:$HOME:/sbin/nologin" >> /etc/passwd
+
+          echo "Setting up ssh bastion"
+          curl https://raw.githubusercontent.com/eparis/ssh-bastion/master/deploy/deploy.sh | bash
+          for i in $(seq 0 60)
+          do
+            BASTION_HOST=$(oc get service -n openshift-ssh-bastion ssh-bastion -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+            if [ ! -z "${BASTION_HOST}" ]; then break; fi
+            sleep 10
+          done
+
+          function bastion_ssh() {
+            retry 10 \
+              ssh -o LogLevel=error -o ConnectionAttempts=100 -o ConnectTimeout=5 -o StrictHostKeyChecking=no \
+                  -o ProxyCommand="ssh -A -o StrictHostKeyChecking=no -o LogLevel=error -o ServerAliveInterval=30 -o ConnectionAttempts=100 -o ConnectTimeout=5 -W %h:%p core@${BASTION_HOST}" \
+                  $@
+          }
+
+          # Machine API won't let the user to destroy the node which runs the controller
+          echo "Finding two masters to destroy"
+          MAPI_POD=$(oc get pod -l k8s-app=controller -n openshift-machine-api --no-headers -o name)
+          SURVIVING_MASTER_NODE=$(oc get ${MAPI_POD} -n openshift-machine-api -o jsonpath='{.spec.nodeName}')
+          mapfile -t MASTER_NODES_TO_REMOVE < <(oc get nodes -l node-role.kubernetes.io/master= -o name | grep -v "${SURVIVING_MASTER_NODE}")
+          MASTER_MACHINES_TO_REMOVE=()
+          for master in ${MASTER_NODES_TO_REMOVE[@]}
+          do
+            MASTER_MACHINES_TO_REMOVE+=($(oc get ${master} -o jsonpath='{.metadata.annotations.machine\.openshift\.io\/machine}' | cut -d '/' -f 2))
+          done
+
+          echo "Destroy two masters"
+          # Scale down etcd quorum guard
+          oc scale --replicas=0 deployment.apps/etcd-quorum-guard -n openshift-machine-config-operator
+
+          for machine in ${MASTER_MACHINES_TO_REMOVE[@]}
+          do
+            retry 10 oc --request-timeout=5s -n openshift-machine-api delete machine ${machine}
+          done
+
+          echo "Confirm meltdown"
+          sleep 30
+          oc --request-timeout=5s get nodes && exit 1
+
+          echo "Upload a script to restore single node cluster"
+          cat > /tmp/install_recovery_scripts.sh <<'SCRIPT_EOF'
+        #!/bin/bash
+        set -x
+        if [[ $EUID -ne 0 ]]; then
+          echo "This script must be run as root"
+          exit 1
+        fi
+
+        PREFIX="https://raw.githubusercontent.com/vrutkovs/openshift-recovery/unify-fixes"
```

Comment:
floating downloads :sob: :wink:

Reply:
Its a temporary measure while we debug MCO scripts.

This is fixed in #3842 which is the same PR + MCO scripts